source/event_stream.c

```diff
@@ -1541,36 +1569,875 @@ napi_value aws_napi_event_stream_client_connection_send_protocol_message(napi_en
+            &napi_activation_message)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_NODEJS_CRT_GENERAL,
+            "id=%p s_aws_extract_activation_options_from_js - failed to get required `message` property from "
+            "activation options",
+            (void *)binding);
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    }
+
+    struct aws_allocator *allocator = aws_napi_get_allocator();
+    if (s_aws_event_stream_message_storage_init_from_js(
+            activation_message, allocator, env, napi_activation_message, binding)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_NODEJS_CRT_GENERAL,
+            "id=%p s_aws_extract_activation_options_from_js - failed to unpack activation message from JS",
+            (void *)binding);
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    }
+
+    return AWS_OP_SUCCESS;
+}
+
+napi_value aws_napi_event_stream_client_stream_activate(napi_env env, napi_callback_info info) {
+
+    struct aws_byte_buf operation_name_buffer;
+    AWS_ZERO_STRUCT(operation_name_buffer);
+    struct aws_event_stream_message_storage activation_message;
+    AWS_ZERO_STRUCT(activation_message);
+
+    napi_value node_args[3];
+    size_t num_args = AWS_ARRAY_SIZE(node_args);
+    napi_value *arg = &node_args[0];
+    AWS_NAPI_CALL(env, napi_get_cb_info(env, info, &num_args, node_args, NULL, NULL), {
+        napi_throw_error(env, NULL, "aws_napi_event_stream_client_stream_activate - Failed to extract parameter array");
+        return NULL;
+    });
+
+    if (num_args != AWS_ARRAY_SIZE(node_args)) {
+        napi_throw_error(env, NULL, "aws_napi_event_stream_client_stream_activate - needs exactly 3 arguments");
+        return NULL;
+    }
+
+    struct aws_event_stream_client_stream_binding *binding = NULL;
+    napi_value node_binding = *arg++;
+    AWS_NAPI_CALL(env, napi_get_value_external(env, node_binding, (void **)&binding), {
+        napi_throw_error(
+            env,
+            NULL,
+            "aws_napi_event_stream_client_stream_activate - Failed to extract stream binding from first "
+            "argument");
+        return NULL;
+    });
+
+    if (binding == NULL) {
+        napi_throw_error(env, NULL, "aws_napi_event_stream_client_stream_activate - binding was null");
+        return NULL;
+    }
+
+    if (binding->is_closed || binding->stream == NULL) {
+        napi_throw_error(env, NULL, "aws_napi_event_stream_client_stream_activate - stream already closed");
+        return NULL;
+    }
+
+    napi_value napi_activation_options = *arg++;
+    if (s_aws_extract_activation_options_from_js(
+            env, napi_activation_options, binding, &operation_name_buffer, &activation_message)) {
+        napi_throw_error(
+            env,
+            NULL,
+            "aws_napi_event_stream_client_stream_activate - unable to unpack activation options from JS object");
+        goto done;
+    }
+
+    napi_value stream_activation_callback = *arg++;
```

Comment:
Question: Does this need to be checked for null?

Reply:
It's synthesized internally, so no.

That being said, I'm going to do an invalid input pass after this with tests and probably some fixes where the tests uncover unhelpful failures.